addons/binding/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/motionsensor/DLinkMotionSensorCommunication.java

```diff
@@ -0,0 +1,434 @@
+                break;
+            case INVALID_PIN:
+                status = DeviceStatus.INVALID_PIN;
+                break;
+            case INTERNAL_ERROR:
+                status = DeviceStatus.INTERNAL_ERROR;
+                break;
+            case UNSUPPORTED_FIRMWARE:
+                status = DeviceStatus.UNSUPPORTED_FIRMWARE;
+                break;
+            case INITIALISED:
+            default:
+                break;
+        }
+    }
+
+    /**
+     * Sends the detection message
+     *
+     * @param isRetry - Has this been called as a result of a login retry
+     * @return true, if the last detection time was successfully retrieved, otherwise false
+     */
+    private boolean getLastDetection(final boolean isRetry) {
+        detectSuccess = false;
+
+        if (loginSuccess) {
+            try {
+                final Document soapResponse = sendReceive(detectionAction, DETECT_TIMEOUT_MS);
+
+                final Node result = soapResponse.getElementsByTagName("GetLatestDetectionResult").item(0);
+
+                if (result != null) {
+                    if (OK.equals(result.getTextContent())) {
+                        final Node timeNode = soapResponse.getElementsByTagName("LatestDetectTime").item(0);
+
+                        if (timeNode != null) {
+                            prevDetection = lastDetection;
+                            lastDetection = Long.valueOf(timeNode.getTextContent());
+                            detectSuccess = true;
+                        } else {
+                            unexpectedResult("getLastDetection - Unexpected response", soapResponse);
+                        }
+                    } else if (isRetry) {
+                        unexpectedResult("getLastDetection - Unexpected response", soapResponse);
+                    }
+                } else {
+                    unexpectedResult("getLastDetection - Unexpected response", soapResponse);
+                }
+            } catch (final Exception e) {
+                // Assume there has been some problem trying to send one of the messages
+                if (status != DeviceStatus.COMMUNICATION_ERROR) {
+                    logger.debug("getLastDetection - Communication error", e);
+                    status = DeviceStatus.COMMUNICATION_ERROR;
+                }
+            }
+        }
+
+        return detectSuccess;
+    }
+
+    /**
+     * Sends the reboot message
+     *
+     */
+    private void reboot() {
+        if (loginSuccess) {
+            try {
+                final Document soapResponse = sendReceive(rebootAction, REBOOT_TIMEOUT_MS);
+
+                final Node result = soapResponse.getElementsByTagName("RebootResult").item(0);
+
+                if (result == null || !OK.equals(result.getTextContent())) {
+                    unexpectedResult("reboot - Unexpected response", soapResponse);
+                }
+            } catch (final Exception e) {
```

Comment:
This was left over from when the code was originally using the SoapConnection classes which did sometimes throw NPE when trying to parse responses during connection/network problems, e.g. turning off the sensor.